### ISSUE TITLE
Is there any way to handle clipboard change event?

### ISSUE THREAD

**Issue by bit-ranger** [1Password/arboard#78]

**Comment by complexspaces** [comment]
Hey there, @bit-ranger. AFAIK, clipboard change events are a feature of window managers and not raw clipboard accesses. Windows, for example, has `AddClipboardFormatListener` that can fire `WM` events for you, but I don't know if its usable in the context of a CLI application off the top of my head.

**Comment by Don-Cryptus** [comment]
its posible https://crates.io/crates/clipboard-master

**Comment by huangganggui** [comment]
@complexspaces  Can i start a thread monitor the `clipboard.get_text()` change or not?

**Comment by complexspaces** [comment]
@huangganggui No, that won't work like you want. `get_text()` does not block waiting for changes, it will either return the current clipboard contents or an error if nothing is available.

In theory: you could read the current clipboard value, poll the clipboard every second in a thread, and see if the value has changed but that's pretty inefficient.

**Comment by huangganggui** [comment]
@complexspaces  Nice!  I will try poll the clipboard every 0.1s in a thread.  In my case, efficiency maybe not such important. 
Thanks a lot.

**Comment by sincekmori** [comment]
A note on a narrower version of this problem, in case it helps. I didn't actually need every clipboard change, just "the user deliberately copied something, grab it". For that shape, polling turns out to be avoidable: you wait for a trigger first, then gate the read on the OS change counter (`NSPasteboard.changeCount` on macOS, the clipboard sequence number on Windows). That also protects you from reading a stale value right after the copy, which bit me more than once.

After rebuilding this in a couple of projects I extracted it into a crate, [copycopy](https://github.com/sincekmori/copycopy). It combines a listen-only Ctrl/Cmd+C+C trigger, the change-counter-gated read, and foreground app info. For genuine "fire on every change" monitoring, clipboard-master mentioned above is still the right tool.
